### ISSUE TITLE
How to use libh2o with blocking socket interface?

### ISSUE THREAD

**Issue by judofyr** [h2o/h2o#261]
Is there a way for me to have a blocking socket reading loop (that I control) and pass the result into libh2o (which will do all the hard HTTP work)?

**Comment by kazuho** [comment]
No.  H2O is tightly coupled to libuv (or the embedded event loop of H2O).

If you are looking for a protocol library that is not bound to a specific event loop library, I would suggest looking at nghttp2 (IIRC HTTP/2 only), or libcurl (IIRC it is capable of speaking HTTP/2 if configured as such).
